lib/Adapter/Git/GitFilesystem.php

```diff
@@ -47,6 +47,11 @@ public function remove($path)
             return parent::remove($path);
         }
 
+        if ($path->isDirectory()) {
+            $this->exec(sprintf('rm -rf %s', $path->path()));
```

Comment:
not rellated to this PR, but this looks dangerous, i'll make a follow up PR to escape this properly.